releasenotes/notes/probabilities_dict_bug_fix-aac3b3d3853828dc.yaml

```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    The internal method :meth:`~.QuantumState._subsystem_probabilities` has been fixed to correctly 
+    order its usage of the ``dims`` attribute. This Fixes a bug in 
+    :meth:`~.QuantumState.probabilities_dict`.
```

Comment:
```suggestion
    Fixed an issue with the :meth:`.Statevector.probabilities_dict` and :meth:`.DensityMatrix.probabilities_dict`
    method where it would return incorrect results for non-qubit systems when the ``qargs`` argument was
    specified.
    Fixed `#9210 <https://github.com/Qiskit/qiskit-terra/issues/9210>`__
```